### PR TITLE
Fixes issue #952 - did not write everything to output stream

### PR DESCRIPTION
**Pull request by cocytus**
Refit might not write everything to the outgoing http stream depending on buffer size and the total amount of data it should sent. This PR fixes that.

Should I also create a PR for 5.2.1, since it's a pretty critical bug, I'd think?

Unsure how to create a good unit/integration test for this - any help would be appreciated.

**Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/reactiveui/refit?pullRequest=1010) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/reactiveui/refit?pullRequest=1010) before we can accept your contribution.<br/><br/>:x: cocytus [sign now](https://cla.dotnetfoundation.org/reactiveui/refit?pullRequest=1010)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/reactiveui/refit?pullRequest=1010) it.</sub>

**Comment by cocytus** [comment]
Note: This also uses WriteAsync as opposed to Write(), which would block the calling thread until everything was written.

**Comment by cocytus** [comment]
Should I instead create a PR to 5.2.1 tag?

**Comment by clairernovotny** [comment]
Thanks! I can cherry pick the result to that branch once it's merged into main.

I would like to get a failing/fixed test that shows this -- the key is figuring out what the interface/datatype/payload was that was causing this? Something that was too big? We serialize a known array of data or a file in the test if that's it.

There's a mocking infrastructure that lets us check the contents of what the server receives, similar to this:
https://github.com/reactiveui/refit/blob/main/Refit.Tests/RestService.cs#L1406-L1429

**Comment by cocytus** [comment]
Test added in separate commit. It fails before last commit is added.

File: Refit/Buffers/PooledBufferWriter.Stream.cs [hunk 1]
```diff
@@ -91,9 +91,12 @@ public override Task CopyToAsync(Stream destination, int bufferSize, Cancellatio
 
                 try
                 {
+#if NETSTANDARD2_1
```
Comment (clairernovotny) [inline Refit/Buffers/PooledBufferWriter.Stream.cs:94]: Does this also need to be `|| NET5_0`?

Comment (cocytus) [reply Refit/Buffers/PooledBufferWriter.Stream.cs]: Thank you, yes. Not for the library to work, but for the tests to actually test this code path.

**Comment by clairernovotny** [comment]
After cherry-picking the fix, there's a failing test on the `rel/v5.2` branch: `CanSerializeBigData`. The `BigData` member is null. Any chance you could take a quick look so we can get fix out for 5.2?

**Comment by cocytus** [comment]
Created PR against 5.2 branch.

**Pull request merged**